Analyzer/plugins/Analyzer.cc

```diff
@@ -2825,79 +2887,270 @@ bool Analyzer::passPreselection(const reco::TrackRef track,
+  
+  if (tuple) {
+    if (tof)
+      tuple->BefPreS_EtaVsTOF->Fill(track->eta(), tof->inverseBeta(), Event_Weight);
+  }
+  
+  if (tuple) {
+    if (GenBeta >= 0)
+      tuple->Beta_PreselectedC->Fill(GenBeta, Event_Weight);
+    if (DZSB && OASB)
+      tuple->BefPreS_Dxy_Cosmic->Fill(dxy, Event_Weight);
+    if (DXYSB && OASB)
+      tuple->BefPreS_Dz_Cosmic->Fill(dz, Event_Weight);
+    if (DXYSB && DZSB)
+      tuple->BefPreS_OpenAngle_Cosmic->Fill(OpenAngle, Event_Weight);
+    
+    // Get the location of the outmost hit
+    const GlobalPoint outerHit = getOuterHitPos(iSetup, dedxHits);
+    const float furthersHitDxy = sqrt(outerHit.x()*outerHit.x()+outerHit.y()*outerHit.y());
+    const float furthersHitDistance = sqrt(outerHit.x()*outerHit.x()+outerHit.y()*outerHit.y()+outerHit.z()*outerHit.z());
+    tuple->BefPreS_LastHitDXY->Fill(furthersHitDxy, Event_Weight);
+    tuple->BefPreS_LastHitD3D->Fill(furthersHitDistance, Event_Weight);
+
+    if (fabs(track->eta()) < DTRegion) {
+      tuple->BefPreS_Pt_DT->Fill(track->pt(), Event_Weight);
+    } else {
+      tuple->BefPreS_Pt_CSC->Fill(track->pt(), Event_Weight);
+    }
+    
+    if (DXYSB && DZSB && OASB) {
+      tuple->BefPreS_Pt_Cosmic->Fill(track->pt(), Event_Weight);
+      tuple->BefPreS_Ias_Cosmic->Fill(Ias, Event_Weight);
+      tuple->BefPreS_Ih_Cosmic->Fill(Ih, Event_Weight);
+    }
+    if (tof) {
+      tuple->BefPreS_TOF->Fill(tof->inverseBeta(), Event_Weight);
+      if (PUA)
+        tuple->BefPreS_TOF_PUA->Fill(tof->inverseBeta(), Event_Weight);
+      if (PUB)
+        tuple->BefPreS_TOF_PUB->Fill(tof->inverseBeta(), Event_Weight);
+      if (dttof->nDof() > 6)
+        tuple->BefPreS_TOF_DT->Fill(dttof->inverseBeta(), Event_Weight);
+      if (csctof->nDof() > 6)
+        tuple->BefPreS_TOF_CSC->Fill(csctof->inverseBeta(), Event_Weight);
+      tuple->BefPreS_PtTOF->Fill(track->pt(), tof->inverseBeta(), Event_Weight);
+    }
+      
+    if (tof) {
+      tuple->BefPreS_TOFIs->Fill(tof->inverseBeta(), Ias, Event_Weight);
+      tuple->BefPreS_TOFIh->Fill(tof->inverseBeta(), Ias, Event_Weight);
+    }
+    
+    //Muon only prediction binned depending on where in the detector the track is and how many muon stations it has
+    //Binning not used for other analyses
+    int bin = -1;
+    if (typeMode_ == 3) {
+      if (fabs(track->eta()) < DTRegion) {
+        bin = muonStations(track->hitPattern()) - 2;
+      } else {
+        bin = muonStations(track->hitPattern()) + 1;
+      }
+      tuple->BefPreS_Pt_Binned[to_string(bin)]->Fill(track->pt(), Event_Weight);
+    }
   }
+
   
   // N-1 plots
   if (tuple) {
-    for (size_t i=0;i<sizeof(passedCutsArray);i++) {
+    for (size_t i=1;i<sizeof(passedCutsArray);i++) {
       bool allOtherCutsPassed = true;
-      for (size_t j=0;j<sizeof(passedCutsArray);j++) {
+      for (size_t j=1;j<sizeof(passedCutsArray);j++) {
         if (i==j) continue;
         if (!passedCutsArray[j]) {
```

Comment:
So the first cut (index=0) is never taken into account for N-1 plots?

Reply:
Yeah, that's the always true part